scripts/plotting/zonal_mean.py

```diff
@@ -263,8 +259,8 @@ def zonal_mean(adfobj):
                         adfobj.add_website_data(plot_name, var, case_name, season=s,
                                                 plot_type="Zonal")
 
-                        #Continue to next iteration:
-                        continue
+                        #Continue to log-p plots:
+                        pass
```

Comment:
I think with this change the non-log-p plots will still be run below even if `redo_plot` is `False`.  I think the best solution would be to move this `redo_plot` check to higher up in the routine (before the data is read in), and to have it check the existence of both the `Zonal_Mean` and `Zonal_Mean_logp` plots, and then to create new logical variables that skip the relevant data-reading and plotting steps as needed depending on if each individual plot type already exists.

Hopefully that makes sense?  If not then we can certainly chat about it in person later this week.

Reply:
@nusbaume Take a look at my most recent commit for the order of checking for plots. 

There is a loop for all possible plot names and for each plot, if redo_plot is false and the file exists, each existing plot name will be added to a list before the data loop.
(For the zonal it has two checks one for regular and one for log-p. It's not the most elegant solution, so I'm open for improvements.)

In the data collection section when it iterates over plot names again before it calculates the seasonal mean, it will check to see if the plot name is in the new list and then calculates the mean if necessary.

My test run with ~10 variables for a single year, when re-running the ADF with existing plots, was a 10X improvement in elapsed time.

Let me know what you think.

Reply:
This update looks great to me, thanks!